assets/shared/helpers/player/use-editor-player.js

```diff
@@ -144,26 +151,33 @@ const useEditorPlayer = ( videoBlock ) => {
 		const playerIframe = doc?.querySelector( 'iframe' );
 		const playerScript = doc?.getElementById( API_SCRIPT_ID );
 
-		// Skip if iframe is not found or player was already added.
-		if ( ! playerIframe || playerScript ) {
+		// Skip if iframe is not found.
+		if ( ! playerIframe ) {
+			return;
+		}
+
+		const setIframePlayer = () => {
+			setPlayer( new Player( playerIframe, w ) );
+		};
+
+		// If player script was already added or loaded.
+		if ( playerScript ) {
+			if ( 'loaded' === playerScript.dataset.loaded ) {
+				setIframePlayer();
+			}
+			playerScript.addEventListener( 'load', setIframePlayer );
+
```

Comment:
I think this logic (related to `playerScript`) could be inside `addScript`, no? Because then we wouldn't need this duplicated logic here. WDYT?

Reply:
Yeah! Much better now: https://github.com/Automattic/sensei/pull/5410/commits/3104fd7362adea39354a3e26034c5423db258ce2